src/test/groovy/com/jvm_bloggers/core/data_fetching/blog_posts/BlogPostFactorySpec.groovy

```diff
@@ -9,6 +9,7 @@ import spock.lang.Subject
 import java.time.LocalDateTime
 import java.time.Month
 
+@Subject(BlogPostFactory)
```

Comment:
What is the advantage of `@Subject` at class level? And why we have to change it from annotation per field as we have it already in the code base?

Reply:
IMHO that is better as test/structure annotations should not interfere with test class contents. It's the same with eg. Hibernate entity mixing on the field column definition, validation conditions and for example JSON parsing exceptions.